dbusmock/templates/networkmanager.py

```diff
@@ -287,4 +473,126 @@ def AddWiFiConnection(self, dev_path, connection_name, ssid_name, key_mgmt):
+    state.
+
+    Please note that this does not set any global properties.
+
+    Returns the new object path.
+    '''
+
+    conn_obj = dbusmock.get_object(connection_device)
+    settings = conn_obj.Get(CSETTINGS_IFACE, 'Settings');
+    conn_uuid = settings['connection']['uuid']
+
+    device_objects = [dbus.ObjectPath(dev) for dev in devices]
+
+    active_connection_path = '/org/freedesktop/NetworkManager/ActiveConnection/' + name
+    self.AddObject(active_connection_path,
+                   ACTIVE_CONNECTION_IFACE,
+                   {
+                       'Devices': device_objects,
+                       'Default6': False,
+                       'Default': True,
+                       'Vpn': False,
+                       'Connection': dbus.ObjectPath(connection_device),
+                       'Master': dbus.ObjectPath('/'),
+                       'SpecificObject': dbus.ObjectPath(specific_object),
+                       'Uuid': conn_uuid,
+                       'State': state,
+                   },
+                   [])
+
+    for dev_path in devices:
+        self.SetDeviceActive(dev_path, active_connection_path)
+
+    active_connections = self.Get(MAIN_IFACE, 'ActiveConnections')
+    active_connections.append(dbus.ObjectPath(active_connection_path))
+    self.SetProperty(MAIN_OBJ, MAIN_IFACE, 'ActiveConnections', active_connections)
+
+    return active_connection_path
+
+
+@dbus.service.method(MOCK_IFACE,
+                     in_signature='ss', out_signature='')
+def RemoveAccessPoint(self, dev_path, ap_path):
+    '''Remove the specified access point.
+
+    You have to specify the device to remove the access point from, and the
+    path of the access point.
+
+    Please note that this does not set any global properties.
+    '''
+
+    dev_obj = dbusmock.get_object(dev_path)
+
+    aps = dev_obj.Get(WIRELESS_DEVICE_IFACE, 'AccessPoints')
+    aps.remove(ap_path)
+    dev_obj.Set(WIRELESS_DEVICE_IFACE, 'AccessPoints', aps)
+
+    dev_obj.access_points.remove(ap_path)
+
+    dev_obj.EmitSignal(WIRELESS_DEVICE_IFACE, 'AccessPointRemoved', 'o', [ap_path])
+
+    self.RemoveObject(ap_path)
+
+
+@dbus.service.method(MOCK_IFACE,
+                     in_signature='ss', out_signature='')
+def RemoveWifiConnection(self, dev_path, connection_path):
+    '''Remove the specified WiFi connection.
+
+    You have to specify the device to remove the connection from, and the
+    path of the Connection.
+
+    Please note that this does not set any global properties.
+    '''
+
+    dev_obj = dbusmock.get_object(dev_path)
+    settings_obj = dbusmock.get_object(SETTINGS_OBJ)
+
+    connections = dev_obj.Get(DEVICE_IFACE, 'AvailableConnections')
+    connections.remove(dbus.ObjectPath(connection_path))
+    dev_obj.Set(DEVICE_IFACE, 'AvailableConnections', connections)
+
+    main_connections = settings_obj.ListConnections()
+    main_connections.remove(connection_path)
```

Comment:
It would be awesome if the connection emitted org.freedesktop.NetworkManager.Settings.Connection.Removed here.